aten/src/ATen/native/cuda/ReduceMomentKernel.cu

```diff
@@ -62,7 +62,35 @@ static void mean_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+template <typename scalar_t, typename acc_t=scalar_t, typename out_t=scalar_t>
+void nanmean_kernel_impl(TensorIterator& iter) {
+  at::Tensor src = iter.tensor(1);
+  at::Tensor dim_prod = at::sum(src.isnan().logical_not());
+  float factor = float(iter.num_output_elements()) / dim_prod.item<float>();
+  gpu_reduce_kernel<scalar_t, out_t>(iter, NanMeanOps<acc_t, float> {factor});
```

Comment:
`nanmean` doesn't give correct results on CUDA because of how `factor` is calculated (to be accurate: how `dim_prod` is calculated). I'm still looking at how this can be resolved.

cc: @mruberry - Would you have anything to suggest here?

Reply:
I'm not totally sure what you're asking - did we resolve this with our offline conversation about computing the number of nan elements in the tensor?

Reply:
Yes, thanks @mruberry! This was resolved with our offline conversation. I'll fix the changes tonight and will then add docs & tests.